### PR TITLE
Signed s3 url

### PR DESCRIPTION
**Pull request by NishanthSpShetty**
# Description

presign s3 url for publicly sharing the resources publicly.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# why
- share user artifacts from the experiments

# How Has This Been Tested?

- manual test involving creation of s3 presigned url

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## dependency
- this change depends on the open issue : #65

### CODE REVIEW

File: proto/netbookai/spawner/spawner.proto [hunk 1]
```diff
@@ -569,3 +571,15 @@ message CopySnapshotResponse {
   string newSnapshotId = 1;
   string newSnapshotUri = 2;
 }
+
+message PresignS3UrlRequest {
+  string region = 1;
+  string accountName = 2;
+  string bucket = 3;
+  string file = 4;
```
Comment (sid255) [inline proto/netbookai/spawner/spawner.proto:579]: Can we call this filePath? As we could be using a wildcard path to get everything in a folder.

File: pkg/service/aws/s3.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	// We need v2 api's, which is out of the scope for the migration
+
+	request.SetContext(ctx)
+	url, err := request.Presign(requestExpireTime)
+	if err != nil {
+		a.logger.Error(ctx, "failed to presign bucket resource", "bucket", req.Bucket, "file", req.File)
+		return nil, errors.Wrap(err, "PresignS3Url")
+	}
+
+	a.logger.Error(ctx, "bucket file signed successfully", "url", url, "bucket", req.Bucket, "file", req.File)
```
Comment (sid255) [inline pkg/service/aws/s3.go:49]: Info level not Error

File: pkg/service/aws/session.go [hunk 3]
```diff
@@ -48,6 +49,7 @@ func getSessionUsingAwsProfile(profile string) (*session.Session, error) {
 
 	sess, err := session.NewSessionWithOptions(
 		session.Options{
+			Config:  aws.Config{Region: aws.String("us-west-2")},
```
Comment (sid255) [inline pkg/service/aws/session.go:52]: Can we get it from config?

**Review by sid255 — CHANGES_REQUESTED** [review]

File: pkg/service/aws/session.go (hunk 3 above)
Comment (NishanthSpShetty) [reply pkg/service/aws/session.go:52]: out of the scope of this PR, another PR addresses this

File: proto/netbookai/spawner/spawner.proto (hunk 1 above)
Comment (NishanthSpShetty) [reply proto/netbookai/spawner/spawner.proto:579]: no, it has to the object key. The object being here the file in the bucket. Let me see if we can specify the folder name

File: pkg/service/aws/s3.go (hunk 2 above)
Comment (NishanthSpShetty) [reply pkg/service/aws/s3.go]: updated.

File: pkg/service/aws/session.go (hunk 3 above)
Comment (NishanthSpShetty) [reply pkg/service/aws/session.go]: merged changes of that PR.

**Review by sid255 — APPROVED** [review]
LGTM

**Pull request merged**